### PR TITLE
feat(privateDb): handle db capabilities

### PR DESCRIPTION
**Pull request by euhmeuh**
### Description of the Change

Only show CRUD buttons of the database page for features that are available in the backend.
To know what features are available, we check the "capabilities" object returned by the API.

### Benefits

We can change the backend and the interface adapts to what is available.

### Possible Drawbacks

Some users might be surprised or confused not to see the expected buttons when using a database that does not support some actions. We may want to tell them why.

### CODE REVIEW

File: client/app/private-database/private-database.service.js [hunk 1]
```diff
@@ -117,8 +117,12 @@ angular.module("services").service(
                         .then((_response) => {
                             database.serviceInfos = _response.data;
                             return database;
-                        })
-                        .catch((err) => this.$q.reject(err.data));
+                        });
+                })
+                .then((db) => {
+                    // reformat capabilities array as an object for easy access
+                    db.capabilities = _.mapKeys(db.capabilities, (c) => c.object);
```
Comment (FredericEspiau) [inline client/app/private-database/private-database.service.js]: `c` could have been named `capability` to enhance readability

File: client/app/private-database/private-database.service.js [hunk 2]
```diff
@@ -117,8 +117,12 @@ angular.module("services").service(
                         .then((_response) => {
                             database.serviceInfos = _response.data;
                             return database;
-                        })
-                        .catch((err) => this.$q.reject(err.data));
+                        });
+                })
+                .then((db) => {
```
Comment (FredericEspiau) [inline client/app/private-database/private-database.service.js]: `db` could have been named `database` but I'm nitpicking

File: client/app/private-database/private-database.service.js [hunk 3]
```diff
@@ -117,8 +117,12 @@ angular.module("services").service(
...
+                    // reformat capabilities array as an object for easy access
```
Comment (FredericEspiau) [inline client/app/private-database/private-database.service.js]: As we had a talk about comments yesterday, I feel like this is the good time to say that this kind of comment feels unnecessary

Comment (euhmeuh) [reply client/app/private-database/private-database.service.js]: I felt like I wanted to explain why I was adding a formatting step to the query, but I guess you're right, it's pretty obvious.

**Pull request merged**